### PR TITLE
fix issue with relation fields param not working (fixes #66)

### LINKED ISSUE #66: relation fields param not limiting as intended
Although the tests pass, the `fields` param is not properly limiting the relationships that are serialized.

The issue with the tests passing seems to be something related to how `_.match` works. When testing an array, it appears that the specified array doesn't need to match exactly. Meaning that the array could have other elements and a match would still be true.

I'll submit a PR to fix the issue.

### PR DESCRIPTION
**Pull request by jamesdixon**

### CODE REVIEW

**Comment by jamesdixon** [comment]
@chamini2 just pushed an updated test for this.

File: src/bookshelf/utils.ts [hunk 1]
```diff
@@ -130,9 +130,11 @@ function relationAllowed(bookOpts: BookOpts, relName: string): boolean {
   let { relations }: BookOpts = bookOpts;
   let { fields }: RelationOpts = relations;
 
-  return relations === true ||
-         relations instanceof Object ||
-         (fields instanceof Array && fields.some((rel: string) => rel === relName));
+  if (fields instanceof Array) {
+      return fields.some((rel: string) => rel === relName);
```
Comment (chamini2) [inline src/bookshelf/utils.ts]: How about:

```
return contains(fields, relName); // contains from lodash
```

Comment (chamini2) [inline src/bookshelf/utils.ts]: Sorry, `includes` they removed `contains` in the new version.

Comment (jamesdixon) [inline src/bookshelf/utils.ts]: Yep. I actually just changed this in the other branch.

Comment (chamini2) [inline src/bookshelf/utils.ts]: So maybe close this PR in favor of the other?

Comment (chamini2) [inline src/bookshelf/utils.ts]: No, this one adds tests...

Comment (jamesdixon) [inline src/bookshelf/utils.ts]: I can add the tests from the other. I accidentally cloned the other branch from this one.

**Comment by jamesdixon** [comment]
Closing in favor of #68

**Pull request closed without merging**